onnx2torch/node_converters/pad.py

```diff
@@ -0,0 +1,126 @@
+__all__ = [
+    'OnnxPadStatic',
+    'OnnxPadDynamic',
+]
+
+from typing import List
+from typing import Optional
+
+import torch
+import torch._C as torch_C
+from torch import nn
+
+from onnx2torch.node_converters.registry import add_converter
+from onnx2torch.onnx_graph import OnnxGraph
+from onnx2torch.onnx_node import OnnxNode
+from onnx2torch.utils.common import OnnxMapping
+from onnx2torch.utils.common import OnnxToTorchModule
+from onnx2torch.utils.common import OperationConverterResult
+from onnx2torch.utils.common import onnx_mapping_from_node
+from onnx2torch.utils.custom_export_to_onnx import CustomExportToOnnx
+from onnx2torch.utils.custom_export_to_onnx import OnnxToTorchModuleWithCustomExport
+
+
+class OnnxPadStatic(nn.Module, OnnxToTorchModule):
+
+    def __init__(
+        self,
+        pads: List[int],
+        mode: str = 'constant',
+        constant_value: float = 0.0,
+    ):
+        super().__init__()
+        self.mode = mode
+        self.pads = pads
+        self.constant_value = constant_value
+
+    def forward(self, input_tensor: torch.Tensor) -> torch.Tensor:
+        return torch.nn.functional.pad(
+            input_tensor,
+            mode=self.mode,
+            pad=self.pads,
+            value=self.constant_value,
+        )
+
+
+class OnnxPadDynamic(nn.Module, OnnxToTorchModuleWithCustomExport):
+
+    def __init__(self, mode: str = 'constant'):
+        super().__init__()
+        self.mode = mode
+
+    @staticmethod
+    def _do_forward(
+        input_tensor: torch.Tensor,
+        mode: str,
+        pads: torch.Tensor,
+        constant_value: Optional[float],
+    ) -> torch.Tensor:
+
+        if constant_value is not None:
+            return torch.nn.functional.pad(input_tensor, mode=mode, pad=pads, value=constant_value)
+
+        return torch.nn.functional.pad(input_tensor, mode=mode, pad=pads)
+
+    def forward(
+        self,
+        input_tensor: torch.Tensor,
+        pads: torch.Tensor,
+        constant_value: Optional[float] = None,
+    ) -> torch.Tensor:
+
```

Comment:
In onnx we have the following modes: `constant`, `reflect`, `edge `.
In pytorch: `constant`, `reflect`, `replicate`, `circular`.
It seems that `edge` and `replicate` are the same. You have to process this. Because if we have `edge` pad in onnx, you will have an error in pytorch.

Reply:
This functional would be implemented in the next pr